dds/DCPS/Serializer.h

```diff
@@ -120,6 +120,7 @@ class OpenDDS_Dcps_Export Encoding {
    */
   enum Kind {
     KIND_CDR_PLAIN = 0x0000,
+    KIND_CDR_PLAIN_ZERO = 0x0001,
```

Comment:
I just saw that one of the XCDR2 KINDs always used zero init padding to true so I tried to follow that model.  I wasn't sure of the complications that would result from it.

Reply:
XDR2 padding should always be zero initialized according to the spec. It actually says XCDR1 padding should always be zero initialized too, but I couldn't decide if we should do that, so I keep the old behavior. Now I'm thinking we should probably just go ahead and do that since it would be the simplest way forward.

Reply:
Do you want me to make that change?

Reply:
Sure. Remove `KIND_CDR_PLAIN_ZERO` and add `zero_init_padding(true);` to the `KIND_CDR_PLAIN` union branch in `Encoding::kind()`.

Reply:
Will do.